### PR TITLE
fix(android): `Utils.dismissKeyboard()` not working with modal

### PR DESCRIPTION
**Pull request by vallemar**
Close: https://github.com/NativeScript/NativeScript/issues/10373

### CODE REVIEW

File: packages/core/utils/android/index.ts [hunk 1]
```diff
@@ -67,7 +68,7 @@ export function dismissSoftInput(nativeView?: android.view.View): void {
 		}
 		windowToken = nativeView.getWindowToken();
 	} else if (getCurrentActivity() instanceof androidx.appcompat.app.AppCompatActivity) {
-		const decorView = getCurrentActivity().getWindow().getDecorView();
+		const decorView = (topmost().modal?.getClosestWindow() ?? getCurrentActivity().getWindow()).getDecorView();
```
Comment (rigor789) [inline packages/core/utils/android/index.ts:71]:
```suggestion
		const decorView = (topmost()?.modal?.getClosestWindow() ?? getCurrentActivity().getWindow()).getDecorView();
```

**Review by rigor789 — CHANGES_REQUESTED** [review]
Small adjustment, in case `topmost()` is undefined.

**Comment by vallemar** [comment]
@rigor789 fix comment and I have changed the code to ensure removing the focus on modals that the root view is not a Page

**Comment by CatchABus** [comment]
@vallemar We have 2 cases of modals. One rendered with `Frame` and another with `Page` or a layout component I think.
See: https://v7.docs.nativescript.org/ui/components/modal-view

Does this fix cover both cases?

**Comment by vallemar** [comment]
@CatchABus ok, I have it working with Frame, Page and other layout as the root view of the modal.
Additionally, I have also realized that the focus was not being removed when it was a modal. Now it is working in all cases.

Also, I think it would be good to make some function to bring us the android currentView and use it in all these places because it is a bit ugly to do this everywhere:
```ts
    const modalDialog = (topmost()?._modalParent ?? topmost())?.modal?._dialogFragment?.getDialog();
    const view = (modalDialog ?? AndroidUtils.getCurrentActivity())
```

**Comment by farfromrefug** [comment]
@vallemar should mention that it wont work in all cases. For example it wont work with android `PopupWindow` or with `@nativescript-community/ui-material-bottomsheet`. The latter uses `_bottomSheetFragment` and not `_dialogFragment` (nor does it defines modal).
Would be nice to find a implementation where plugins could register for ways to get current windowToken.
I even had a worse case with  `PopupWindow`. I wanted to hide the keyboard when the  `PopupWindow` got hidden. But the `dismiss` event is fired after the  `PopupWindow` view is removed from tree and thus my view `windowToken` is null :s, there is no way around it :s.
My only option was to use a deprecated method to hide keyboard which does not need a windowToken https://github.com/Akylas/NativeScript/commit/825863be9784da7af298a8758a475c9801ca7940

**Comment by vallemar** [comment]
@farfromrefug I understand the problem, and if something common should be done so that the plugins can do this, otherwise we will continue to have this problem with new plugins or views that do not come from NS-core. Maybe you could make a queue of foreground views so you can bring up the last one and grab the current `windowToken`.

@rigor789  While we are thinking about how to do this, I think another PR could be made, since this solves a current problem in the core and I think it should be merged.

**Comment by farfromrefug** [comment]
@vallemar yes it is a question of another PR! 
I LL start thinking about it. But if anyone has an idea dont hesitate ;)

**Comment by rigor789** [comment]
Maybe we need some sort of "fragment stack" to keep track of fragments... but I can see that getting out-of-sync with reality easily. Or maybe we don't need the fragments, just the windowTokens - in which case, a similar `push/pop` stack could be useful as long as it's updated correctly by core/plugins.

**Review by NathanWalker — APPROVED** [review]

**Pull request merged**